javascript/trees/tree.test.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { BinarySearchTree } = require('./index');
+
+describe ('Binary search trees', () => {
+    it('should instantiate an empty tree', () => {
+        const tree = new BinarySearchTree();
+        expect(tree.root).toBeNull();
+    });
+
+    it('should instantiate a tree with a single root node', () => {
+        const tree = new BinarySearchTree();
+        tree.add(3);
+        expect(tree.root).not.toBeNull();
+        expect(tree.root.value).toBe(3);
+    });
+
+    it('should add left and right child nodes properly', () => {
+        const tree = new BinarySearchTree();
+        tree.add(3);
+        tree.add(6);
+        tree.add(9);
```

Comment:
We probably want to add another value that's less than our initial value of 3 so we can test adding to the left branch...